src/google/protobuf/util/time_util.h

```diff
@@ -37,7 +37,10 @@
 #include <ostream>
 #include <string>
 #ifdef _MSC_VER
-#include <winsock2.h>
+struct timeval {
```

Comment:
The thing that makes me hesitate about this is that if someone happens to include this header and also winsock2.h then there will be an error about the struct being redefined. Is there a way we could restrict this change so that it applies to XBox One only?

Reply:
That's certainly a good point of concern.

Yes, we could guard it with `#ifdef _XBOX_ONE`, though I was hesitant to do that because there are other problems that need to be addressed to get pb compiling for Xbox One (such as all of io_win32.cc needing to be skipped, some filesystem calls like `access()` and friends not being supported, etc.). It felt like adding that include guard would send a signal that the platform was supported which isn't true even after this change.

Perhaps a `GOOGLE_PROTOBUF_` define could be added to control this instead, though that may also open a can of worms.

Reply:
Guarding it with `_XBOX_ONE` sounds good to me.

Reply:
Replaced this commit with an `_XBOX_ONE` guard and added a second commit necessary to get the library compiling for Xbox One.